### ISSUE TITLE
kcov: error: Cannot open linux-vdso.so.1

### ISSUE THREAD

**Issue by gsingh93** [SimonKagstrom/kcov#26]
I get this error when running `kcov` on Arch Linux:
```
kcov: error: Cannot open linux-vdso.so.1
```

`ldd` shows the shared library exists. The kcov folder is generated, but the HTML shows that the binary was never run.

Any ideas?

**Comment by remram44** [comment]
I'm running into this.
```
remram@missy:/tmp$ cat > test.c
int main(int argc, const char *argv[])
{
    return 0;
}
remram@missy:/tmp$ gcc -g test.c -o test
remram@missy:/tmp$ kcov test_kcov ./test
Error: Cannot open linux-vdso.so.1

remram@missy:/tmp$
```
I'm running v27 built today on Debian Jessie x86_64.

**Comment by remram44** [comment]
Note that surprisingly, this doesn't prevent the generation of a report for the C program (although the Rust programs gets 0% reported coverage).

**Comment by SimonKagstrom** [comment]
Hmm... And in your case, both programs are 64-bit binaries?

If I run a 32-bit program on a 64-bit system (with a 32-bit kcov), I see the linux-vdso.so.1 problem as well.

Do you see anything special if you run kcov under strace? Note that the coverage collection will not work in that case, but I think the linux-vdso.so.1 issue happens at startup. Reopening for now.

**Comment by remram44** [comment]
Everything is 64bit here. I can [reproduce this on Vagrant](https://gist.github.com/remram44/d5e443af0296c83522ba) so you should have no problem troubleshooting it.

[Here is the result from strace anyway](https://gist.github.com/remram44/7219f5e0100b59b9a94c)

I don't think this is what prevents me from getting coverage reports from my Rust program though, since I get the report for a C program... I must be missing something else.

**Comment by SimonKagstrom** [comment]
Unfortunately, I don't know rust Is there something akin to -g missing from the compilation command line there? If you run kcov with --debug=15 you'll get debug output for e.g., where it tries to set breakpoints and where they are hit afterwards.

I'll take a look at the issue again. I didn't know about vagrant before, looks useful, thanks!

**Comment by SimonKagstrom** [comment]
I'm working on reproducing this - having had some problems with Vagrant. I'm hoping I can look at this tonight though.

**Comment by SimonKagstrom** [comment]
Narrowing it down for a bit, this is caused by the solib handling. This is done through using a LD_PRELOAD:ed library, and if this is omitted through running kcov with --skip-solibs, the warning/error goes away.

I suspect this will not help with your rust issue though.

**Comment by remram44** [comment]
I got Rust code to work fine, on both Ubuntu (where the error is not shown) and Debian (where the error is still shown) :smiley:

So definitely not a critical issue, although since it always happens you probably want to do something about it.

**Comment by SimonKagstrom** [comment]
Fixed with b61c93e.

It turns out that this was just a spurious issue anyway. kcov tries to parse linux-vdso.so.1 since it's part of the memory map (as a loaded library). However, this library is just inserted by the kernel, and is not present in the file system and it can't be opened because of that. So the fix just turns the error message into a debug warning.
